src/App.tsx

```diff
@@ -12,13 +18,109 @@ const goodsFromServer: string[] = [
   'Honey',
   'Jam',
   'Garlic',
-];
+].map(good => ({
+  name: good,
+  id: uuidv4(),
+}));
+
+export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortBy, setSortBy] = useState('none');
+  const [isVisible, setVisible] = useState(false);
+
+  const visibilityApp = () => {
+    setVisible(true);
+  };
+
+  const getVisibilityGoods = () => {
+    const copyGoods = [...goodsFromServer];
+
+    const preperedCreater = copyGoods.sort((firstGood, secondGood) => {
+      switch (sortBy) {
+        case 'name':
+          return firstGood[sortBy].localeCompare(secondGood[sortBy]);
+        case 'length':
+          return firstGood.name.length - secondGood.name.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      copyGoods.reverse();
+    }
+
+    return preperedCreater;
+  };
+
+  const reverseGoods = () => {
+    setReversed(current => !current);
+  };
+
+  const sortByName = () => {
+    setSortBy('name');
+  };
+
+  const sortByLength = () => {
+    setSortBy('length');
+  };
+
+  const resetGoods = () => {
+    setSortBy('none');
+    setReversed(false);
+  };
+
```

Comment:
can use `useMemo` here
```suggestion
	const visibleGoods  = getVisibilityGoods()
```